src/main/java/org/opensearch/knn/index/KNNSettings.java

```diff
@@ -106,6 +108,11 @@ public class KNNSettings {
     public static final String KNN_DEFAULT_VECTOR_STREAMING_MEMORY_LIMIT_PCT = "1%";
 
     public static final Integer ADVANCED_FILTERED_EXACT_SEARCH_THRESHOLD_DEFAULT_VALUE = -1;
+    public static final Integer KNN_DEFAULT_QUANTIZATION_STATE_CACHE_SIZE_LIMIT_PERCENTAGE = 5; // By default, set aside 5% of the JVM for
+                                                                                                // the limit
+    public static final Integer KNN_MAX_QUANTIZATION_STATE_CACHE_SIZE_LIMIT_PERCENTAGE = 10; // Quantization state cache limit cannot exceed
+                                                                                             // 10% of the JVM heap
+    public static final Integer KNN_DEFAULT_QUANTIZATION_STATE_CACHE_EXPIRY_TIME_MINUTES = 60;
```

Comment:
what is reasoning behind these limits?

Reply:
I added initial ones based on conversation with @Vikasht34 .  We'll probably need to think more on this with benchmarking but for now I needed a value to put

Reply:
Here is the reasoning behind this 

Number of segments :- 2500 ( this must be exterme) 
Dimesion 16000 
Binary Quantization 4 bit 
Threshould will looks like this 
```
thresholds[0] {0.1f, 0.2f, 0.3f  upto 16k }
     *   thresholds[1] {0.4f, 0.5f, 0.6f upto 16k}
     *   thresholds[2] {0.7f, 0.8f, 0.9f upto 16k}
     *   thresholds[3] {1.0f, 1.1f, 1.2f upto 16k}
``` 
For 2500 segments, the total size of the thresholds array would be:
2500 segments x 16,000 columns x 4 bytes/element = 1,600,000,000 bytes
Converting this to more familiar units:

    1,600,000,000 bytes = 1.6 GB (gigabytes)

So, the total size of your thresholds array for 2500 segments would be approximately 1.6 GB.
If you have a 32 GB heap, this array would occupy:
(1.6 GB / 32 GB) x 100% ≈ 5%

this is extreme case assuming all vectors are 16000 size and 2500 segments 

Assuming P90 case where avregae Dimesion is 5000 with 4bit :- 0.5 GB which 1.5625% of 32gb heap , so we choose 5% as safe limit which can handle exterme case as well.

Reply:
>  2500 segments x 16,000 columns x 4 bytes/element = 1,600,000,000 bytes

This is 160,000,000 bytes - 150 MB. 5k is 50 MB. That being said, I dont think we need these limits.

Reply:
Also, if we do need the limits, can we not make them settings? Settings are hard to remove. It seems in the future, we may want to rely on IndexInput and mmap to handle caching. Id prefer not to expose these as settings